modules/packages/LinearAlgebra.chpl

```diff
@@ -221,6 +222,267 @@ class LinearAlgebraError : Error {
     }
 }
 
+/*
+This class promotes lazy computation of required
+Helper matrices to evaluate matrix exponential.
+
+High precision of expm requires smaller computations,
+hence we do a lazy computation depending on the norm value
+of the matrix.
+
+*/
+class ExpmPadeHelper {
+  type t;
+  var A : t;
+  var A2 : t;
+  var A4 : t;
+  var A6 : t;
+  var A8 : t;
+  var A10 : t;
+  var Exactd4 : real;
+  var Exactd6 : real;
+  var Exactd8 : real;
+  var Exactd10 : real;
+  var Approxd4 : real;
+  var Approxd6 : real;
+  var Approxd8 : real;
+  var Approxd10 : real;
+  var ident : t;
+  var use_exact_onenorm : bool;
+  // Boolean arrays to know if A, Exactd, Approxd
+  // have been computed.
+  var isAComputed : [1..5] bool;
+  var isExactdComputed : [1..4] bool;
+  var isApproxdComputed : [1..4] bool;
+
+  /*
+  Params:
+  A: Expects an N*N square matrix
+  use_exact_onenorm: boolean value specifying if the onenorm has to be exact.
```

Comment:
If we do intend to document this class, you'll want to follow the syntax that makes for pretty generated documentation ([The UnitTest module](https://github.com/chapel-lang/chapel/blob/master/modules/packages/UnitTest.chpl#L769) can be used as an example and you can find a more explicit description [in the chpldoc documentation](https://chapel-lang.org/docs/latest/tools/chpldoc/chpldoc.html))